lib/instrumentation/modules/aws-sdk/s3.js

```diff
@@ -65,6 +85,30 @@ function instrumentationS3 (orig, origArguments, request, AWS, agent, { version,
     return orig.apply(request, origArguments)
   }
 
+  if (params) {
+    const attrs = {}
+    // I guess we have to check that params correspond to operation
+    // because maybe consumers can append params arbitrarily
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/90007a192ae8b06d40ee8c15f56256d69a330321/semantic_conventions/trace/instrumentation/aws-sdk.yml#L448
+    if (params.Bucket && request.operation !== 'listBuckets') {
+      attrs['aws.s3.bucket'] = bucket
+    }
+
+    if (params.Key && OPERATIONS_WITH_KEY.indexOf(request.operation) !== -1) {
+      attrs['aws.s3.key'] = request.params.Key
+    }
+
+    // span._getOTelAttributes() initialises the object so we have to make sure there is
+    // something to put inside and not increase the span payload for nothing
+    const attrKeys = Object.keys(attrs)
+    if (attrKeys.length > 0) {
+      const otelAttrs = span._getOTelAttributes()
+      attrKeys.forEach(function (key) {
+        otelAttrs[key] = attrs[key]
+      })
+    }
+  }
```

Comment:
In the scope of this change makes a lot of sense. `Key` cannot be present if there is no `Bucket`. Also thanks for the optimisation for the operation guard. I was hesitant to add it because

- we will have to maintain that list of allowed operations for `Key` param
- I was wondering if we may want to get wrongly passed params to operations. Like the example you showed for `listBuckets` operation in previous comment (it will log the failure and report the params in the span)

Reply:
Update: David and I discussed on a call. We decided to not bother with limiting looking for a `params.Key` to an explicit set of operation names. Rather David will make the instrumentation add the bucket and key attributes if the params are there -- even if they are there in error.  This is what the Python agent is doing FWIW.